src/geometry/CMakeLists.txt

```diff
@@ -7,10 +7,30 @@ file(GLOB SOURCES "*.cpp")
 add_library(redisearch-geometry STATIC ${SOURCES})
 
 # include_directories(${absl_INCLUDE_DIR})
+message(STATUS "BOOST_DIR: ${BOOST_DIR}")
 
-if(NOT BOOST_DIR STREQUAL "")
+if(NOT BOOST_DIR STREQUAL "" AND EXISTS ${BOOST_DIR})
  	message(STATUS "BOOST_DIR is not empty: ${BOOST_DIR}")
 	include_directories(${BOOST_DIR})
 else()
- 	message(STATUS "BOOST_DIR is not defined")
+ 	message(STATUS "BOOST_DIR is not defined or empty")
+		# set(BOOST_INCLUDE_LIBRARIES boost geometry optional unordered)
+	set(BOOST_ENABLE_CMAKE ON)
+
+	message(STATUS "fetching boost")
+
+	include(FetchContent)
+	FetchContent_Declare(
+	Boost
+	GIT_REPOSITORY https://github.com/boostorg/boost.git
+	GIT_TAG boost-1.84.0
```

Comment:
Should the actual boost version be set in the Makefile? (users could see it more easily if they wish to do it on their own)